plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -69,14 +69,25 @@ def set_secrets
       expand_secret_annotations
     end
 
-    def images
-      all = containers.reject { |c| c[:'samson/dockerfile'] == 'none' }
+    def build_selectors
+      all = containers
       modify_init_container { |containers| all += containers }
-      all.map { |c| c.fetch(:image) }.uniq
+      all.map { |c| build_selector_for_container(c) }.compact
     end
 
     private
 
+    def build_selector_for_container(container)
+      if project.docker_image_building_disabled?
+        # TODO: weird that there is no opt-out here
+        [nil, container.fetch(:image)]
+      else
+        selected = container[:"samson/dockerfile"] || 'Dockerfile'
+        next if selected == 'none'
+        [selected, nil]
```

Comment:
Would this break if someone has their dockerfile as `Dockerfile` and specifies an image name on their build? Then they would have a build with a [nil docker file](https://github.com/zendesk/samson/blob/master/app/models/build.rb#L76) and we wouldn't be able to find a matching build.

Reply:
yes, builds with nil `dockerfile` column cannot be found when not using `external images` feature, but that's a separate discussion/feature ... trying to keep this refactoring only :)